MapboxVision/src/main/java/com/mapbox/vision/location/android/AndroidLocationEngineImpl.kt

```diff
@@ -8,71 +8,74 @@ import android.location.LocationManager
-import com.mapbox.vision.location.LocationEngineListener
+import com.mapbox.vision.mobile.core.interfaces.LocationEngineListener
+import com.mapbox.vision.utils.VisionLogger
 
 internal class AndroidLocationEngineImpl(context: Context) : LocationEngine, LocationListener {
 
     private val locationManager: LocationManager = context.getSystemService(Context.LOCATION_SERVICE) as LocationManager
     private val currentProvider: String = LocationManager.GPS_PROVIDER
 
-    private lateinit var locationEngineListener: LocationEngineListener
+    private var locationEngineListener: LocationEngineListener? = null
 
-    // Bearing returned can be invalid - it'l contain 0.0f then. We cache last valid bearing to return it to subscribers.
-    private var lastValidBearing : Float = 0f
-
-    private var attached = false
+    // Bearing returned can be invalid - it'll contain 0.0f then. We cache last valid bearing to return it to subscribers.
+    private var lastValidBearing: Float = 0f
 
     @SuppressLint("MissingPermission")
     override fun attach(locationEngineListener: LocationEngineListener) {
         this.locationEngineListener = locationEngineListener
         if (!TextUtils.isEmpty(currentProvider)) {
             locationManager.requestLocationUpdates(currentProvider, DEFAULT_MIN_TIME, DEFAULT_MIN_DISTANCE, this)
         }
-        attached = true
     }
 
     override fun detach() {
-        attached = false
+        locationEngineListener = null
         locationManager.removeUpdates(this)
     }
 
-    override fun isAttached(): Boolean = attached
-
-    // Location Engine listener
     override fun onLocationChanged(location: Location) {
-        if (!::locationEngineListener.isInitialized) {
-            return
-        }
-
-        if (location.bearing != 0.0f) {
-            lastValidBearing = location.bearing
-        }
+        locationEngineListener?.let { listener ->
+            if (location.bearing != 0.0f) {
+                lastValidBearing = location.bearing
+            }
 
-        locationEngineListener.onNewLocation(
+            listener.setLocation(
                 latitude = location.latitude,
                 longitude = location.longitude,
                 speed = location.speed,
-                altitude = location.altitude,
+                altitude = location.altitude.toFloat(),
                 horizontalAccuracy = location.accuracy,
                 verticalAccuracy = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) location.verticalAccuracyMeters else 0.0F,
                 bearing = lastValidBearing,
                 timestamp = location.time
-        )
+            )
+        }
     }
 
     override fun onStatusChanged(provider: String?, status: Int, extras: Bundle?) {
-        Log.v(TAG, String.format("Provider %s status changed to %d (current provider is %s).",
-                provider, status, currentProvider))
+        VisionLogger.v(
+            TAG,
+            String.format(
```

Comment:
kotlin style :) 
`"Provider $provider status changed to $status (current provider is $currentProvider)."
`
In 3 places

Reply:
Lets do this in separate PR, it's just dev-master sync

Reply:
sounds good